lib/puppet/transaction/resource_harness.rb

```diff
@@ -92,6 +93,25 @@ def perform_changes(resource, context)
     end
 
     capture_audit_events(resource, context)
+    persist_system_values(resource, context)
+  end
+
+  # We persist the last known values for the properties of a resource after resource
+  # application.
+  # @param [Puppet::Type] resource resource whose values we are to persist.
+  # @param [ResourceApplicationContent] context the application context to operate on.
+  def persist_system_values(resource, context)
```

Comment:
I assume this will persist the actual file contents in a situation like this?

```puppet
file { '/tmp/foo': content => 'really_long_string', }
```